AssetRipperCore/Classes/Shader/Shader.cs

```diff
@@ -176,7 +182,49 @@ public override IEnumerable<PPtr<IUnityObjectBase>> FetchDependencies(Dependency
 
 		protected override YAMLMappingNode ExportYAMLRoot(IExportContainer container)
 		{
-			throw new NotSupportedException();
+			YAMLMappingNode node = ExportBaseYAMLRoot(container);
+			node.InsertSerializedVersion(ToSerializedVersion(container.ExportVersion));
+			node.Add("m_ParsedForm", m_ParsedForm.ExportYAML(container));
+			node.Add("platforms", Array.ConvertAll(Platforms, value => (int) value).ExportYAML(false));
+			if (!IsSerialized(container.Version))
+			{
+				node.Add("decompressedSize", DecompressedSize);
+
+			}
+			else if (IsDoubleArray(container.Version))
+			{
+				node.Add("offsets", Offsets2D.ExportYAML(false));
+				node.Add("compressedLengths", CompressedLengths2D.ExportYAML(false));
+				node.Add("decompressedLengths", DecompressedLengths2D.ExportYAML(false));
+			}
+			else
+			{
+				node.Add("offsets", Offsets1D.ExportYAML(false));
+				node.Add("compressedLengths", CompressedLengths1D.ExportYAML(false));
+				node.Add("decompressedLengths", DecompressedLengths1D.ExportYAML(false));
+			}
+			node.Add("compressedBlob", CompressedBlob.ExportYAML());
+			if (HasDependencies(container.Version))
+			{
+				node.Add("m_Dependencies", Dependencies.ExportYAML(container));
+			}
+			if (HasNonModifiableTextures(container.Version))
+			{
+				node.Add("m_NonModifiableTextures", NonModifiableTextures.ExportYAML(container));
+			}
+			if (HasShaderIsBaked(container.Version))
+			{
+				node.Add("m_ShaderIsBaked", ShaderIsBaked);
+			}
+			//Editor-Only
+			node.Add("m_DefaultTextures", (new Dictionary<string, PPtr<Texture>>()).ExportYAML(container));
+			node.Add("m_CompileInfo", (new ShaderCompilationInfo()).ExportYAML(container));
+			if (HasCompileSmokeTest(container.Version))
+			{
+				node.Add("m_CompileSmokeTestAfterImport", default(bool));
+			}
+
+			return node;
```

Comment:
Since AssetRipper exports at a minimum of 2017.3.0, I think this should be fine. Otherwise, it wouldn't be.